rdm/project_management/base.py

```diff
@@ -0,0 +1,6 @@
+class BaseBackend:
+    def __init__(self, config):
+        self.config = config
+
+    def pull(self):
+        raise NotImplementedError()
```

Comment:
I am curious if you have an opinion on using `NotImplementedError` vs. creating an abstract base class and using the `@abstractmethod` decorator for methods like these. Either approach seems fine; I noticed that you don't get a pylint error about unimplemented abstract methods until you try and instantiate the class, so I guess there isn't much development life improvement by using one over the other.

Reply:
Good idea! Fixed!